### PR TITLE
Add recipe for PEXSI

### PR DESCRIPTION
**Pull request by awvwgk**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by awvwgk** [comment]
@conda-forge/staged-recipes This recipe is ready for review. It packages a C++ MPI library for Linux and OSX. Currently only works with MPICH since superlu_dist is not available with OpenMPI.

**Comment by awvwgk** [comment]
Just talked to the PEXSI maintainers. Seems like I can't do much more than building for MPICH and SuperLU_dist 6.2.0, while waiting for a new release of PEXSI here.

@conda-forge/staged-recipes Gentle ping for a review and maybe a merge of this recipe.

File: recipes/pexsi/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://bitbucket.org/berkeleylab/{{ name }}/downloads/{{ name }}_v{{ version }}.tar.gz
+  sha256: 8bfad6ec6866c6a29e1cc87fb1c17a39809795e79ede98373c8ba9a3aaf820dd
+  patches:
+    - superlu_dist-cname.patch
+    - cmake-shared-lib.patch
```
Comment (ocefpaf) [inline recipes/pexsi/meta.yaml:15]: Are these upstream in a PR or a commit? If so, please add a link/comment here. If not, are they upstreamable? Then consider sending them to the source. (Not a blocker to merge here.)

File: recipes/pexsi/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+build:
+  number: {{ build }}
+  skip: True  # [win]
+  script:
+    - export CXX="$PREFIX/bin/mpicxx" CC="$PREFIX/bin/mpicc" FC="$PREFIX/bin/mpifort"
+    - cmake ${CMAKE_ARGS} -DBUILD_SHARED_LIBS=ON -DPEXSI_ENABLE_OPENMP=ON -G Ninja -B _build
+    - cmake --build _build
+    - cmake --install _build
```
Comment (ocefpaf) [inline recipes/pexsi/meta.yaml:24]: Is there a ctest/make check we can run here?

File: recipes/pexsi/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    - {{ compiler('cxx') }}
+    - {{ compiler('fortran') }}
+    - cmake
+    - ninja
+    - llvm-openmp  # [osx]
+    - libgomp  # [linux]
+  host:
+    - {{ mpi }}
+    - parmetis
+    - superlu_dist 6.2.0
```
Comment (ocefpaf) [inline recipes/pexsi/meta.yaml:44]: Probably worth pinning in our central pinning.

Comment (awvwgk) [reply recipes/pexsi/meta.yaml:46]: Central pinning is at 7.* or similar, I hope I can drop the explicit pin here for the next version when upstream switches to superlu_dist 7.

File: recipes/pexsi/meta.yaml (hunk 1 above)
Comment (awvwgk) [reply recipes/pexsi/meta.yaml:15]: Upstream is updating for a newer superlu_dist version already, but no release yet unfortunately. I'll stay in touch with upstream.

**Pull request merged**